uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -0,0 +1,1208 @@
+        return buildIdpWithAliasProperties(idzId, aliasId, aliasZid, originKey, UAA);
+    }
+
+    private void arrangeAliasFeatureEnabled(final boolean enabled) {
+        ReflectionTestUtils.setField(identityProviderEndpoints, "aliasEntitiesEnabled", enabled);
+    }
+
+    private static IdentityProvider<?> buildIdpWithAliasProperties(
+            final String idzId,
+            final String aliasId,
+            final String aliasZid,
+            final String originKey,
+            final String type
+    ) {
+        final AbstractIdentityProviderDefinition definition = buildIdpDefinition(type);
+
+        final IdentityProvider<AbstractIdentityProviderDefinition> provider = new IdentityProvider<>();
+        provider.setIdentityZoneId(idzId);
+        provider.setAliasId(aliasId);
+        provider.setAliasZid(aliasZid);
+        provider.setName(originKey);
+        provider.setOriginKey(originKey);
+        provider.setType(type);
+        provider.setConfig(definition);
+        provider.setActive(true);
+        return provider;
+    }
+
+    private static AbstractIdentityProviderDefinition buildIdpDefinition(final String type) {
+        switch (type) {
+            case OIDC10:
+                final OIDCIdentityProviderDefinition definition = new OIDCIdentityProviderDefinition();
+                try {
+                    return definition
+                            .setAuthUrl(new URL("https://www.example.com/oauth/authorize"))
+                            .setLinkText("link text")
+                            .setRelyingPartyId("relying-party-id")
+                            .setRelyingPartySecret("relying-party-secret")
+                            .setShowLinkText(true)
+                            .setSkipSslValidation(true)
+                            .setTokenKey("key")
+                            .setTokenKeyUrl(new URL("https://www.example.com/token_keys"))
+                            .setTokenUrl(new URL("https://wwww.example.com/oauth/token"));
+                } catch (final MalformedURLException e) {
+                    throw new RuntimeException(e);
+                }
+            case UAA:
+                final PasswordPolicy passwordPolicy = new PasswordPolicy();
+                passwordPolicy.setExpirePasswordInMonths(1);
+                passwordPolicy.setMaxLength(100);
+                passwordPolicy.setMinLength(10);
+                passwordPolicy.setRequireDigit(1);
+                passwordPolicy.setRequireUpperCaseCharacter(1);
+                passwordPolicy.setRequireLowerCaseCharacter(1);
+                passwordPolicy.setRequireSpecialCharacter(1);
+                passwordPolicy.setPasswordNewerThan(new Date(System.currentTimeMillis()));
+                return new UaaIdentityProviderDefinition(passwordPolicy, null);
+            default:
+                throw new IllegalArgumentException("IdP type not supported.");
+        }
+    }
+}
```

Comment:
There appears to be good coverage here, though I wonder if something is missed. I would prefer id there was a doc or matrix indicating conditions that maps to the tests to validate if any gaps exist.

Reply:
There is a way to see the test coverage on the new code [in SonarCloud](https://sonarcloud.io/summary/new_code?id=cloudfoundry-identity-parent&pullRequest=2637). Overall coverage on new code is reported as 80.7% currently, just over the minimum threshold.

If I drill into IdentityProviderEndpoints.java, I see some `if` statement bodies with non-trivial code that would be nice to get test coverage for, plus some exceptions that aren't tested.

Also, Sonar indicates that there is one new code smell introduced in the PR that I'd like to see addressed.

Reply:
I resolved the Sonar finding with e5f5791 and added some unit tests to increase the coverage with fafe033.